core/commonMain/src/kotlinx/serialization/encoding/AbstractDecoder.kt

```diff
@@ -74,8 +74,7 @@ public abstract class AbstractDecoder : Decoder, CompositeDecoder {
         index: Int,
         deserializer: DeserializationStrategy<T?>,
         previousValue: T?
-    ): T? {
-        val isNullabilitySupported = deserializer.descriptor.isNullable
-        return if (isNullabilitySupported || decodeNotNullMark()) decodeSerializableValue(deserializer, previousValue) else decodeNull()
+    ): T? = decodeIfNullable(deserializer) {
+        decodeSerializableElement(descriptor, index, deserializer, previousValue)
```

Comment:
Why change `decodeSerializableValue` to `decodeSerializableElement`?

Reply:
not sure anymore honestly. I can't come up with a reason, I guess it just "felt right" because it took all the parameters. I reverted that.